sponsorship_switzerland/reports/sponsorships_evolution_months_report.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Sebastien Toth <popod@me.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import tools
+from odoo import models, fields, api
+
+
+class SponsorshipsEvolutionMonthsReport(models.Model):
+    _name = "sponsorships.evolution_months.report"
+    _description = "Sponsorships Evolution By Months"
+    _auto = False
+    _rec_name = 'start_date'
+
+    start_date = fields.Char(string="start date", readonly=True)
+    cumulative_sponsorships = fields.Integer(string="cumulative sponsorships",
+                                             readonly=True)
+
+    @api.model_cr
+    def init(self):
+        tools.drop_view_if_exists(self.env.cr,
+                                  'sponsorships_evolution_months_report')
+        self.env.cr.execute("""
+            CREATE OR REPLACE VIEW sponsorships_evolution_months_report AS
+            SELECT
+              sub.start_date AS start_date,
+              ROW_NUMBER() OVER (ORDER BY (SELECT 100)) AS id,
+              SUM(sub.total) over (ORDER BY sub.start_date)
+                AS cumulative_sponsorships
+            FROM (
+              SELECT
+                to_char(date_trunc('month', rc.start_date),'YYYY.MM')
+                AS start_date, COUNT(rc.start_date) AS total
+              FROM recurring_contract AS rc
+              WHERE rc.state = 'active'
```

Comment:
Replace this filter by `activation_date is not null` because we should include all sponsorship that are no longer active now but were active in the past.

Reply:
yes right.